pkg/addonmanager/manager.go

```diff
@@ -81,12 +87,50 @@ func (a *addonManager) Trigger(clusterName, addonName string) {
 }
 
 func (a *addonManager) Start(ctx context.Context) error {
-	kubeClient, err := kubernetes.NewForConfig(a.config)
+	var addonNames []string
+	for key := range a.addonAgents {
+		addonNames = append(addonNames, key)
+	}
+
+	// To support sending ManifestWorks to different drivers (like the Kubernetes apiserver or MQTT broker), we build
+	// ManifestWork client that implements the ManifestWorkInterface and ManifestWork informer based on different
+	// driver configuration.
+	// Refer to Event Based Manifestwork proposal in enhancements repo to get more details.
+	_, config, err := cloudeventswork.NewConfigLoader(a.options.WorkDriver, a.options.WorkDriverConfig).
```

Comment:
could we move all this cloudEvent setting to a separated func instead of the main Start func? or a separated cloudEvent manager? I am concerned about this change on the existing addon-framework breaking existing addons.

Reply:
Implemented `cloudeventsAddonManager` in separate package `cloudevents`, so that we don't break existing addons. see: https://github.com/open-cluster-management-io/addon-framework/pull/256/commits/dcf9539ec9090eb372454e2c626fe4528364fce7